add-extension.js

```diff
@@ -109,13 +110,14 @@ Alternative usage: node add-extension --download=VSIX_URL`);
         }
         location = path.dirname(locations[0]);
     }
+    const packagePath = path.join('/tmp/repository', location, 'package.json')
     /** @type {{ publisher: string, name: string, version: string }} */
-    const package = JSON.parse(await readFile(path.join('/tmp/repository', location, 'package.json'), 'utf-8'));
-    ['publisher', 'name', 'version'].forEach(key => {
-      if (!(key in package)) {
-        throw new Error(`Expected "${key}" in ${location}/package.json: ${JSON.stringify(package, null, 2)}`);
-      }
-    });
+    const package = JSON.parse(await readFile(packagePath, 'utf-8'));
+    if (registry.requiresLicense && !(await ovsx.isLicenseOk(packagePath, package))) {
+      throw new Error(`License must be present, please ask author of extension to add license (${repository})`)
+    } else {
+      ovsx.validateManifest(package)
```

Comment:
we have to validate the manifest if license is not required, otherwise manifest is validate by _isLicenseOk_ method.